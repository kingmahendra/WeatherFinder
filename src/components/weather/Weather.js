import React from 'react'

export const Weather = props => (
        <div className="weather__info"> 
            {
                props.city && props.country && <p className="weather__key"> Loction :
                  <span className="weather__value"> {props.city} {props.country}  </span>
                </p> 
            }
            {
                props.weather.temperature && <p className="weather__key"> Temperature : 
                    <span className="weather__value"> {props.weather.temperature} &#8451; </span>
                </p> 
            }
            {
                props.weather.humidity && <p className="weather__key"> Humidity : 
                    <span className="weather__value"> {props.weather.humidity} &#37; </span>
                </p>
            }
            {
                props.weather.windspeed &&  <p className="weather__key"> Wind Speed : 
                 <span className="weather__value"> {props.weather.windspeed} Km/h</span>
                </p>
            }
            {
                props.weather.description && <p className="weather__key"> Condition : 
                    <span className="weather__value"> {props.weather.description} </span>
                </p>
            }
            {
                props.error && <p className="weather__error">{props.error}</p>
            }
          
        </div>
     )
