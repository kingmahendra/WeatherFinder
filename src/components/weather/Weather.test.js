import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'

import { Weather } from './Weather'

describe('Wheather Component', ()=> {
    it('should render city and country and weather info', () => {
        const state = {
            city: 'London',
            country: 'UK',
            weather: {
               temperature: 31,
               humidity: 10,
               windspeed: 10,
               description: 'some desc',
            },
            error: ''
           }
        const component = shallow(<Weather city={state.city} country={state.country} weather={state.weather} error={state.error}/>)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
    })

    it('should render error', () => {
        const state = {
            city: undefined,
            country: undefined,
            weather:{
              temperature: undefined,
              humidity: undefined,
              windspeed: undefined,
              description: undefined,
            },
            error: 'City not found'
        }
        const component = shallow(<Weather city={state.city} country={state.country} weather={state.weather} error={state.error}/>)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
    })


})

