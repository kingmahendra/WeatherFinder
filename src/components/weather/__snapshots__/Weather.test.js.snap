// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wheather Component should render city and country and weather info 1`] = `
<div
  className="weather__info"
>
  <p
    className="weather__key"
  >
     Loction :
    <span
      className="weather__value"
    >
       
      London
       
      UK
        
    </span>
  </p>
  <p
    className="weather__key"
  >
     Temperature :
    <span
      className="weather__value"
    >
       
      31
       
    </span>
  </p>
  <p
    className="weather__key"
  >
     Humidity :
    <span
      className="weather__value"
    >
       
      10
       
    </span>
  </p>
  <p
    className="weather__key"
  >
     Wind Speed :
    <span
      className="weather__value"
    >
       
      10
       
    </span>
  </p>
  <p
    className="weather__key"
  >
     Condition :
    <span
      className="weather__value"
    >
       
      some desc
       
    </span>
  </p>
</div>
`;

exports[`Wheather Component should render error 1`] = `
<div
  className="weather__info"
>
  <p
    className="weather__error"
  >
    City not found
  </p>
</div>
`;
